### PR TITLE
Simplify the prebuild script

### PR DESCRIPTION
**Pull request by tk-nguyen**
Detect existence of `pubspec.yaml` instead of hardcoding module names.

### CODE REVIEW

**Review by tddang-linagora — APPROVED** [review]

File: scripts/prebuild.sh [hunk 1]
```diff
@@ -4,36 +4,11 @@ set -e
...
-
-cd ../server_settings
-flutter pub get && dart run build_runner build --delete-conflicting-outputs
-
-cd ..
-flutter pub get \
-    && dart run build_runner build --delete-conflicting-outputs \
-    && dart run intl_generator:extract_to_arb --output-dir=./lib/l10n lib/main/localizations/app_localizations.dart \
-    && dart run intl_generator:generate_from_arb --output-dir=lib/l10n --no-use-deferred-loading lib/main/localizations/app_localizations.dart lib/l10n/intl*.arb
+flutter pub get
```
Comment (dab246) [inline scripts/prebuild.sh:7]: `flutter pub get` for each module is necessary. Running it in the main module will likely not update other modules.

Comment (tk-nguyen) [reply scripts/prebuild.sh]: Readded.

File: scripts/prebuild.sh [hunk 2]
```diff
@@ -4,36 +4,10 @@ set -e
 # debug log
 set -x
 
-cd core
-flutter pub get && dart run build_runner build --delete-conflicting-outputs
+find . -name "pubspec.yaml" -exec bash -c '
+  (cd $(dirname "$0") && flutter pub get && dart run build_runner build -d)
+' {} \;
```
Comment (hoangdat) [inline scripts/prebuild.sh:9]: please keep the order of execute build-runner

Comment (tk-nguyen) [reply scripts/prebuild.sh:9]: Can't, only way to do this is hardcoding

Comment (dab246) [reply scripts/prebuild.sh:9]: Screenshot 2024-10-09 at 10 01 45

IMO, we should specify the modules then run the loop.

Comment (tk-nguyen) [reply scripts/prebuild.sh:9]: > Screenshot 2024-10-09 at 10 01 45
> 
> IMO, we should specify the modules then run the loop.

Yes, this is my original plan

**Comment by dab246** [comment]
- [ ] Squash it to one commit

**Comment by tk-nguyen** [comment]
> * [ ]  Squash it to one commit

Done.

**Review by hoangdat — APPROVED** [review]

**Pull request merged**